tests/unittests/CoreGraphics.drawing/CGContextDrawing_ImageMaskTests.cpp

```diff
@@ -0,0 +1,314 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//****************************************************************************** 
+
+#include "DrawingTest.h"
+#include "DrawingTestConfig.h"
+#include "ImageHelpers.h"
+#include <windows.h>
+
+#include <unordered_map>
+#include <vector>
+#include <functional>
+
+enum ClippingShape {
+    ClippingShapeDiamond = 0x10,
+    ClippingShapeSquare = 0x20,
+    ClippingShapeRectangle = 0x30,
+};
+
+enum ClippingType {
+    ClippingTypeMask = 0x0,
+    ClippingTypeAlpha = 0x1,
+};
+
+template <typename TLambda>
+// TLambda is a function of type void(CGContextRef, CGSize, ClippingType).
+CGImageRef __CreateClipImage(CGSize size, ClippingType clipType, TLambda&& drawFunction) {
+    size_t bitsPerColor = 8;
+    size_t componentsPerPixel = clipType == ClippingTypeAlpha ? 4 : 1; // RGBA or G8
```

Comment:
Could, but then somebody would say:

> nit: the values here are non-obvious.

---
In reply to: [92093869](https://github.com/Microsoft/WinObjC/pull/1528#discussion_r92093869) [](ancestors = 92093869)